src/main/java/org/broadinstitute/hellbender/tools/exome/ReadCountCollectionUtils.java

```diff
@@ -312,12 +408,235 @@ public static String getSampleNameFromReadCounts(final ReadCountCollection count
                 writer.writeRecord(new ReadCountRecord.SingleSampleRecord(new Target(interval), coverage));
             }
         } catch (final IOException e) {
-            throw new UserException.CouldNotCreateOutputFile(outFile, e);
+            throw new UserException.CouldNotCreateOutputFile(outName, e);
         }
 
+    }
 
+    /**
+     * Impute zero counts to the median of non-zero values in the enclosing target row.
+     *
+     * <p>The imputation is done in-place, thus the input matrix is well be modified as a result of this call.</p>
+     *
+     * @param readCounts the input and output read-count matrix.
+     */
+    public static void imputeZeroCountsAsTargetMedians(final ReadCountCollection readCounts, final Logger logger) {
+
+        final RealMatrix counts = readCounts.counts();
+        final int targetCount = counts.getRowDimension();
+
+        final Median medianCalculator = new Median();
+        int totalCounts = counts.getColumnDimension() * counts.getRowDimension();
+
+        // Get the number of zeroes contained in the counts.
+        final long totalZeroCounts = IntStream.range(0, targetCount)
+                .mapToLong(t -> DoubleStream.of(counts.getRow(t))
+                        .filter(c -> c == 0.0).count()).sum();
```

Comment:
The filter statement is part of the `DoubleStream`, not the `IntStream`, so it shouldn't align, right?